wp-includes/class-wp-https-ui.php

```diff
@@ -0,0 +1,233 @@
+	 * Adds a settings field to the 'Reading Settings' page.
+	 */
+	public function add_settings_field() {
+		add_settings_field(
+			self::SETTING_ID,
+			__( 'HTTPS', 'pwa' ),
+			array( $this, 'render_settings' ),
+			self::OPTION_GROUP
+		);
+	}
+
+	/**
+	 * Renders the HTTPS settings in /wp-admin on the Reading Settings page.
+	 */
+	public function render_settings() {
+		$upgrade_https_value           = (bool) get_option( self::UPGRADE_HTTPS_OPTION );
+		$upgrade_insecure_content      = (bool) get_option( self::UPGRADE_INSECURE_CONTENT_OPTION );
+		$https_more_details            = sprintf(
+			'<a href="%s">%s</a>',
+			__( 'https://make.wordpress.org/support/user-manual/web-publishing/https-for-wordpress/', 'pwa' ),
+			esc_html__( 'More details', 'pwa' )
+		);
+		$insecure_content_more_details = sprintf(
+			'<a href="%s">%s</a>',
+			__( 'https://developer.mozilla.org/en-US/docs/Web/Security/Mixed_content/How_to_fix_website_with_mixed_content', 'pwa' ),
+			esc_html__( 'More details', 'pwa' )
+		);
+
+		/**
+		 * Todo: change this description based on what insecure content is found.
+		 *
+		 * Have a separate description for when there's only passive insecure content.
+		 * And one for when there's active and passive insecure content.
+		 *
+		 * @see https://github.com/xwp/pwa-wp/issues/17#issuecomment-406188455
+		 */
+		/* translators: %s: a link for more details */
+		$insecure_content_description = esc_html__( 'Your home page doesn&#8217;t contain insecure URLs. However, there may be URLs on other pages that could be blocked. %s', 'pwa' );
+
+		/*
+		 * Todo: change ! get_option() to get_option, as this is only like this for development.
+		 * The WP_HTTPS_Detection doesn't work with my local SSL certificate, and always returns false.
+		 * Also, change $this->is_currently_https() to ! $this->is_currently_https().
+		 * This is also for development only.
+		 * This main if block should only run if the site can use HTTPS, but the 'home' and 'siteurl' options aren't HTTPS.
+		 */
+		if ( $this->is_currently_https() && ! get_option( WP_HTTPS_Detection::HTTPS_SUPPORT_OPTION_NAME ) ) :
+			?>
+			<p class="description">
+				<?php
+				echo wp_kses_post( sprintf(
+					/* translators: %s: a link for more details */
+					esc_html__( 'HTTPS is essential to securing your WordPress site, we strongly suggest upgrading to HTTPS on your site. %s', 'pwa' ),
+					$https_more_details
+				) );
+				?>
+			</p>
+			<p style="margin-top: 20px;"><strong><?php esc_html_e( 'HTTPS Upgrade', 'pwa' ); ?></strong></p>
+			<p>
+				<label><input name="<?php echo esc_attr( self::UPGRADE_HTTPS_OPTION ); ?>" type="radio" <?php checked( $upgrade_https_value ); ?> value="<?php echo esc_attr( self::OPTION_SELECTED_VALUE ); ?>"><?php esc_html_e( 'Yes', 'pwa' ); ?></label>
+				<label><input name="<?php echo esc_attr( self::UPGRADE_HTTPS_OPTION ); ?>" type="radio" <?php checked( ! $upgrade_https_value ); ?> value="0" ><?php esc_html_e( 'No', 'pwa' ); ?></label>
```

Comment:
I think a checkbox should be used instead of radio buttons. I checked the various admin screens and I don't see there being any examples of Yes/No radio buttons being used. Radio buttons seem to only be used in core when there are more than two options. (The exception here is the Yes/No toggles in the Publish metabox in the classic editor, but even those radio buttons are going away in favor of a toggle in Gutenberg.)

With this done, then to detect for the value going into the `self::UPGRADE_HTTPS_OPTION` option, it should be `isset( $_POST[self::UPGRADE_HTTPS_OPTION] )` as opposed to looking at the value.

Reply:
Yeah, checkboxes seem more natural for Yes/No options. There was some discussion in [https://github.com/xwp/pwa-wp/issues/17#issuecomment-404330851](https://github.com/xwp/pwa-wp/issues/17#issuecomment-404330851) about the radio buttons showing that this is a reversible decision.

Reply:
Why would a checkbox imply that it is not a reversible decision?

Reply:
I think it's still clear with a checkbox that it's a reversible decision.

Reply:
Let's go with checkboxes then for consistency with the rest of the admin settings screens. /cc @jwold

Reply:
Sounds good, I'll make a commit for that.

Reply:
[https://github.com/xwp/pwa-wp/pull/36/commits/c8da8580e50d33205032fd6fc4ddd04b9615c3a7](https://github.com/xwp/pwa-wp/pull/36/commits/c8da8580e50d33205032fd6fc4ddd04b9615c3a7) changes the radio buttons to checkboxes:

<img width="1264" alt="checkboxes" src="https://user-images.githubusercontent.com/4063887/43346278-287d47b8-91b6-11e8-81b2-007903d262cb.png">

It also [changes the validation](https://github.com/xwp/pwa-wp/commit/c8da8580e50d33205032fd6fc4ddd04b9615c3a7#diff-cd06676b91b7dea56c702cb7bc9e7ec9R103) callback to return `true` if `$_POST` value `isset()`, like:

```php
if ( isset( $_POST[ self::UPGRADE_HTTPS_OPTION ] ) ) { // WPCS: CSRF OK.
	return true;
}
```

If that's not what you had in mind in [https://github.com/xwp/pwa-wp/pull/36#discussion_r205622089](https://github.com/xwp/pwa-wp/pull/36#discussion_r205622089), I'll change this.